File: Brackets/Brackets.java

public class Brackets {	
	public static boolean isValid(String textToCheck) {
	    int textLength = textToCheck.length();
	    int topPosition = 0;
	    int stackSize = 0;
	    char[] stack = new char[textLength];
	    
	    if (textLength == 0) {
	        return true;
	    }
	    
	    if (textLength % 2 != 0) {
	        return false;
	    }
	    
	    for (char c : textToCheck.toCharArray()) {	        
	        if (c == '(' || c == '[' || c == '{') {
	            stack[topPosition] = c;
	            stackSize++;
	        } else {
	            if (c == ')') {
	                if (stack[topPosition] != '(') {
	                    return false;
	                }
	            }
	            else if (c == ']') {
	                if (stack[topPosition] != '[') {
	                    return false;
	                }
	            }
	            else if (c == '}') {
	                if (stack[topPosition] != '{') {
	                    return false;
	                }
	            }
                
	            stackSize--;
	        }
	        
	        if (stackSize > 0) {
	            topPosition = stackSize - 1;
	        } else {
	            topPosition = 0;
	        }
	        
	    }
	    
	    return true;
	}
}

File: Brackets/brackets.py
def is_valid(s):
    """
    :type s: str
    :rtype: bool
    """
    s_length = len(s)

    if s_length == 0:
        return True

    if s_length % 2 != 0:
        return False

    size = 0
    stack = []

    for char in s:
        if char == '(' or char == '[' or char == '{':
            stack.append(char)
        else:
            if char == ')':
                if stack[size] != '(':
                    return False
            elif char == ']':
                if stack[size] != '[':
                    return False
            elif char == '}':
                if stack[size] != '{':
                    return False
            
            stack.pop()
        size = len(stack) - 1

    return True
